proxy/http/HttpSessionManager.cc

```diff
@@ -98,7 +113,10 @@ ServerSessionPool::acquireSession(sockaddr const *addr, INK_MD5 const &hostname_
     // Otherwise we need to scan further matches to match the host name as well.
     // Note we don't have to check the port because it's checked as part of the IP address key.
     if (TS_SERVER_SESSION_SHARING_MATCH_IP != match_style) {
-      while (loc && loc->hostname_hash != hostname_hash) {
+      while (loc) {
+        if (loc->hostname_hash == hostname_hash && match_sni(sm, loc->get_netvc())) {
```

Comment:
I don't think this is the right approach. ``match_sni`` should do exactly what it's name claims and we should check in the caller whether we are looking for a TLS session. I think that having ``match_sni`` make assumptions about the match is spooky action at a distance that makes this harder to understand than is necessary.